### PR TITLE
Docker things

### PR DESCRIPTION
**Pull request by edqx**
Added Dockerfile, added .dockerignore and changed license to the proper GPL3 expression.

Only for the backend, not sure what you want to do about the front end but I can do that too.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM node:14
+
+EXPOSE 8079
+
+WORKDIR /usr/src/app
+
+COPY package.json .
+RUN yarn install
```
Comment (cybershard) [inline Dockerfile:8]: This should be moved under the `COPY . .` instruction.
```suggestion
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+FROM node:14
+
+EXPOSE 8079
+
+WORKDIR /usr/src/app
+
+COPY package.json .
```
Comment (cybershard) [inline Dockerfile:7]: Remove this and just do one copy instruction, or combine the copy instructions into one
```suggestion
```

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,13 @@
+FROM node:14
+
+EXPOSE 8079
```
Comment (cybershard) [inline Dockerfile:3]: Move it to the next build stage, under the second `FROM node:14`

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+WORKDIR /usr/src/app
+
+COPY package.json .
+RUN yarn install
+
+COPY . .
+RUN yarn build
+
```
Comment (cybershard) [inline Dockerfile:12]: Add another `FROM` instruction to create the next build stage.
```suggestion
FROM node:14
ENV NODE_ENV=production
WORKDIR /usr/src/app-prod
COPY /usr/src/app/package.json .
RUN yarn install
COPY /usr/src/app/dist ./dist/
```

**Review by cybershard — CHANGES_REQUESTED** [review]

File: package.json [hunk 5]
```diff
@@ -6,12 +6,14 @@
...
-  "license": "GPLv3",
+  "license": "GPL-3.0-only",
   "scripts": {
     "serve": "nodemon ./src/main.ts",
     "build": "cross-env NODE_ENV=production tsc -p ./",
+    "build:docker": "docker build -t auproxy-backend .",
+    "start": "node ./dist/src/main.js",
     "lint": "eslint ./src/**/*.ts --fix",
-    "heroku-postbuild": "yarn build && YARN_PRODUCTION=false yarn --cwd auproximity-webui install && yarn --cwd auproximity-webui build && mkdir dist/src/dist && cp -r auproximity-webui/dist dist/src"
+    "build:all": "yarn build && YARN_PRODUCTION=false yarn --cwd auproximity-webui install && yarn --cwd auproximity-webui build && mkdir dist/src/dist && cp -r auproximity-webui/dist dist/src"
```
Comment (cybershard) [inline package.json:16]: The change from `heroku-postbuild` to `build:all` should be reverted, because we aren't going with the `build:all` route

**Review by cybershard — CHANGES_REQUESTED** [review]

File: package.json (hunk 5 above)
Comment (edqx) [reply package.json:16]: ok fair enough.

**Review by cybershard — APPROVED** [review]

**Pull request merged**